platform/ios/src/MGLMapView.mm

```diff
@@ -3003,10 +3069,10 @@ - (void)locationManager:(__unused CLLocationManager *)manager didUpdateLocations
 
     if (! oldLocation || ! CLLocationCoordinate2DIsValid(oldLocation.coordinate) || [newLocation distanceFromLocation:oldLocation])
     {
-        self.userLocation.location = newLocation;
-
-        // deselect user if applicable since we don't do callout tracking yet
-        if ([self.selectedAnnotation isEqual:self.userLocation]) [self deselectAnnotation:self.userLocation animated:NO];
+        if (self.userTrackingState != MGLUserTrackingStateBegan)
+        {
+            self.userLocation.location = newLocation;
```

Comment:
MapKit does a simple, easeTo-style animation to the first location, which makes it easier to accommodate intervening updates. By contrast, this branch uses a Google-style flyTo, which would be easy to modify mid-flight even after implementing #3610.

In their own ways, both MapKit and Google Maps favor context and continuity over up-to-the-second precision. If a particular application needs to go immediately to the actual user location, it can pass NO into -setUserTrackingMode:animated:, which is being made public in this PR.